examples/deploy/terraform/cluster/main.tf

```diff
@@ -25,13 +25,54 @@ module "eks" {
   bastion_info        = local.infra.bastion
   create_eks_role_arn = local.infra.create_eks_role_arn
   tags                = local.infra.tags
+}
+data "aws_caller_identity" "global" {
+  provider = aws.global
+}
+
+data "aws_caller_identity" "this" {}
+
+locals {
+  # Determine if the EKS cluster is in the same account as the hosted zone
+  is_eks_account_same = data.aws_caller_identity.this.account_id == data.aws_caller_identity.global.account_id
+}
 
+# If you are enabling the IRSA configuration for external-dns.
+# You will need to add the role created(module.irsa_external_dns.irsa_role) to
+# the following annotation to the `external-dns` service account:
+# `eks.amazonaws.com/role-arn: <<module.irsa_external_dns.irsa_role>>`
+module "irsa_external_dns" {
+  count               = var.irsa_external_dns != null && var.irsa_external_dns.enabled ? 1 : 0
+  source              = "./../../../../modules/eks/submodules/irsa"
+  use_cluster_odc_idp = local.is_eks_account_same
+  eks_info            = module.eks.info
+  external_dns        = var.irsa_external_dns
+
+  providers = {
+    aws = aws.global
+  }
+}
+
+module "irsa_policies" {
+  count                   = var.irsa_policies != null ? 1 : 0
+  source                  = "./../../../../modules/eks/submodules/irsa"
+  use_cluster_odc_idp     = true
+  eks_info                = module.eks.info
+  additional_irsa_configs = var.irsa_policies
 }
 
+# Provider configuration for the account where the hosted zone is defined.
+# Useful in configurations where accounts do not have a public hosted zone(i.e us-gov regions) and internet routing(public DNS)
+# is instead defined in a different account. Configure the `global` aws alias accordingly,
+# by specifying the profile belonging to the account pertaining to the hosted zone.
 provider "aws" {
-  region = var.infra.region
+  alias = "global"
```

Comment:
this is a bit weird.
you are aliasing it as global but not setting it up based on the aws.global provider.

Reply:
we do not have the dual credentials set up, as a matter of fact we have not even discussed on how to do that yet. the idea is for the current session to be implicit(works for both gov/commercial) and the `global` to be explicit. If they are the same which is the case for commercial deployments then we toggle the `use_cluster_odc_idp` to `true` if they are not then `false` to create it using the `global` provider